src/inc/quic_platform_winuser.h

```diff
@@ -782,7 +818,16 @@ CxPlatEventQEnqueue(
     sqe->IsQueued;
 #endif
     CxPlatZeroMemory(&sqe->Overlapped, sizeof(sqe->Overlapped));
-    return PostQueuedCompletionStatus(*queue, 0, 0, &sqe->Overlapped) != 0;
+    if (!PostQueuedCompletionStatus(*queue, 0, 0, &sqe->Overlapped)){
+        if (sqe->WcpEvent){
```

Comment:
I don't think using a wait completion packet should ever be a backup: it should be an OR. If we created the SQE with the intent of using a wait completion we should always use it.

Reply:
What I was trying to say above is that we don't want to call `PostQueuedCompletionStatus`, and to fall back to `WcpEvent` if it fails: we need to know which of the two control the event (based on how it was setup) and call the right one and only that one.

Having our own event queue + a single trigger event might be a good way to simplify it (but will take a bigger refactor, with the risk of re-inventing the wheel?).

I think we can keep this one PR scope on the smaller side by focusing on how to queue the event in the IOCP: we can start by using wait completion events for specific events only. When that works, we can have follow up with our own queue to avoid useless queuing in KM (might be worth perf measurements).